GymBuddies/ViewControllers/ProfileViewController.m

```diff
@@ -7,41 +7,214 @@
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<UIImagePickerControllerInfoKey,id> *)info{
+    NSURL *urlImage = [info objectForKey:UIImagePickerControllerImageURL];
+    NSString *imageName = urlImage.lastPathComponent;
+    
+    UIImage *editedImage = [info objectForKey:UIImagePickerControllerEditedImage];
+    PFFileObject *imageFile = [ParseAPIManager getPFFileFromImage:editedImage imageName:imageName];
+    
+    [ParseAPIManager changeProfilePicture:imageFile completion:^(BOOL succeeded, NSError * _Nonnull error) {
+        if(succeeded){
+            self.userProfilePicture.image = [info objectForKey:UIImagePickerControllerEditedImage];
         } else {
-            SceneDelegate *delegate = (SceneDelegate *)self.view.window.windowScene.delegate;
-            UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
-            delegate.window.rootViewController = [storyboard instantiateViewControllerWithIdentifier:@"LoginNavigationController"];
+            UIAlertController *alert = [AlertCreator createOkAlert:@"Error changing your profile picture" message:error.localizedDescription];
+            [self presentViewController:alert animated:YES completion:nil];
         }
     }];
+    
+    [self dismissViewControllerAnimated:YES completion:nil];
 }
 
-- (void)viewDidLoad {
-    [super viewDidLoad];
-    // Do any additional setup after loading the view.
+
+#pragma mark - Fetching routines
+
+-(void)fetchUsersCreatedRoutines{
+    [ParseAPIManager fetchUsersCreatedRoutines:^(NSArray * _Nonnull elements, NSError * _Nullable error) {
+            if(elements != nil){
+                self.createdRoutineList = elements;
+                [self.routinesTableView reloadData];
+            } else{
+                UIAlertController *alert = [AlertCreator createOkAlert:@"Error loading your routines" message:error.localizedDescription];
+                [self presentViewController:alert animated:YES completion:nil];
+            }
+    }];
+}
+
+
+-(void)fetchUsersLikedRoutines{
+    [ParseAPIManager fetchUsersLikedRoutines:^(NSArray * _Nonnull elements, NSError * _Nullable error) {
+            if(elements != nil){
+                self.likedRoutineList = elements;
+                [self.routinesTableView reloadData];
+            } else{
+                UIAlertController *alert = [AlertCreator createOkAlert:@"Error loading your liked routines" message:error.localizedDescription];
+                [self presentViewController:alert animated:YES completion:nil];
+            }
+    }];
+}
+
+
+#pragma mark - Button interaction
+
+- (IBAction)didTapCreated:(id)sender {
+    if(self.showCreatedRoutinesButton.isSelected){
+        return;
+    }
+    self.showCreatedRoutinesButton.selected = YES;
+    self.showLikedRoutinesButton.selected = NO;
+    self.showCreatedOrLikedRoutinesIndicator = kShowCreatedRoutines;
+   
+    [self.routinesTableView reloadData];
```

Comment:
refactor this -- as well
`didTapCanceled` and `didTapLiked` have exactly the same logic.